debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -204,8 +204,26 @@ public synchronized void start(Configuration config) {
                     throw new ConnectException(
                             "The MySQL server does not appear to be using a row-level binlog, which is required for this connector to work properly. Enable this mode and restart the connector.");
                 }
-                // We're going to start by reading the binlog ...
-                chainedReaderBuilder.addReader(binlogReader);
+
+                // if there are new tables
+                if (newTablesInConfig()) {
+                    // and we are configured to run a parallel snapshot
+                    if (taskContext.snapshotNewTables() == MySqlConnectorConfig.SnapshotNewTables.PARALLEL) {
+                        ParallelSnapshotReader parallelSnapshotReader = new ParallelSnapshotReader(config, taskContext, getNewFilters(offsets, config));
+                        ReconcilingBinlogReader reconcilingBinlogReader = parallelSnapshotReader.createReconcilingBinlogReader();
+                        MySqlTaskContext unifiedTaskContext = createAndStartTaskContext(config, getAllFilters(config));
+                        // take any final steps requires for reconciling the parallel readers
+                        reconcilingBinlogReader.uponCompletion(new CompleteReconciliation(unifiedTaskContext, reconcilingBinlogReader, source, config));
+                        BinlogReader unifiedBinlogReader = new BinlogReader("binlog", unifiedTaskContext, null);
+                        chainedReaderBuilder.addReader(parallelSnapshotReader);
+                        chainedReaderBuilder.addReader(reconcilingBinlogReader);
+                        chainedReaderBuilder.addReader(unifiedBinlogReader);
+                    }
+                } else {
```

Comment:
Aren't we missing to set up the binlog reader in case there _are_ new tables but snapshotNewTables is set to `OFF`? Seems reasonable to let it stream the previously configured tables.

Reply:
oops, you're right. I changed the flow here and didn't even think about that.